competition/evaluation/copy_data.py

```diff
@@ -10,7 +10,10 @@ def __init__(self):
         self._agent_names = None
 
     def __call__(self, **kwargs):
-        self._data = copy.deepcopy(dict(**kwargs))
+        try:
+            self._data = copy.deepcopy(dict(**kwargs))
+        except RecursionError:
+            self._data = copy.copy(dict(**kwargs))
```

Comment:
`deepcopy` was used to prevent external users from modifying some of the `infos` objects later using additional env wrappers.   

How does `RecursionError` occur and would a shallow copy be a sufficient tamper protection here?

Reply:
I am unsure how the `RecursionError` occurred but some part of the SMARTS `Observation` type causes it. I intend to fix this during the competition.

Reply:
https://github.com/huawei-noah/SMARTS/issues/1540

Reply:
@Adaickalavan To be clear, I think this is fine to update because it is only useful to us.